### DISCUSSION TITLE
Add Compare API in GraphQL

### DISCUSSION THREAD

**Discussion by sashaafm** [community/community#11358] [Other Feature Feedback, Questions, & Ideas]
Hello GitHub,

I'd like to ask (or suggest) if there are plans to include a similar operation of comparing two commits to the GraphQL API like it's present in the REST API one? The action in question is the following: [Compare API](https://docs.github.com/en/rest/reference/commits#compare-two-commits).

This would greatly benefit integrators using this API as the GraphgQL version would reduce a lot of traffic usage and machine resources as the unwanted patches could be omitted from response.

Thank you!

**Comment by sashaafm** [comment]
An alternative I'd like to suggest is to add parameters that somehow allow unwanted patches from being omitted in the response.

**Comment by jarhill0** [comment]
On [refs](https://docs.github.com/en/graphql/reference/objects#ref), it's possible to compare two branches/tags (e.g. comparing `myFeatureBranch` to `main`):

```graphql
query { 
  repository(name: "my_repo", owner: "me") { 
    ref(qualifiedName: "main") {
      compare(headRef: "myFeatureBranch") {
        commits(first: 5) {
          nodes {
            oid
          }
        }
      }
    }
  }
}
```

This gives all of the commits on `myFeatureBranch` that are not on `main`.

_BUT_, when trying to pass in a commit SHA, it doesn't work, because both `ref#qualifiedName` and `compare#headRef` expect [git References](https://git-scm.com/book/en/v2/Git-Internals-Git-References), and do not accept commit SHAs.

As I understand it, a git ref is pretty much just a nice label that resolves to a specific commit SHA, so it seems weird to me that we cannot pass a SHA directly here.

**Reply by whitecat** [reply]
Sadly this does not help when the branch is merged to `main`, there will be no difference between the two and you cannot recover the branch commits.
